samtranslator/model/events.py

```diff
@@ -16,3 +20,18 @@ class EventsRule(Resource):
     }
 
     runtime_attrs = {"rule_id": lambda self: ref(self.logical_id), "arn": lambda self: fnGetAtt(self.logical_id, "Arn")}
+
+    def __init__(
+        self,
+        logical_id: Optional[Any],
+        relative_id: Optional[str] = None,
+        depends_on: Optional[List[str]] = None,
+        attributes: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        super().__init__(logical_id, relative_id, depends_on, attributes)
+
+        if len(self.logical_id) > EVENT_RULE_ID_MAX_LENGTH:
+            # According to documentation, event rule id has max length of 64 characters limit
+            # https://docs.aws.amazon.com/eventbridge/latest/APIReference/API_Target.html
+            # Truncate logical id to 59 characters and append 'Event' as a workaround
+            self.logical_id = self.logical_id[:59] + "Event"
```

Comment:
```suggestion
            suffix = "Event"
            self.logical_id = self.logical_id[:(64-len(suffix))] + suffix
```
Might be a bit cleaner

Reply:
`suffix` can be a module constant or class `Final` attribute

Reply:
Or make it a function. It seems like a fairly generic operation for something like SAM.

```python
def truncate_suffix(length: int, suffix: Optional[str]):
```

Reply:
^ `truncate_with_max_length_and_add_a_suffix_if_the_input_exceeds_the_max_length()`